D2L.Hypermedia.Siren.Tests/SirenActionTests.cs

```diff
@@ -175,6 +176,96 @@ public class SirenActionTests {
 			TestHelpers.ArrayBidirectionalEquality( actions, others, false );
 		}
 
+		private static ISirenAction[] HashCodeActions()
+        {
+			return new ISirenAction[]
+            {
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" )
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					@class: new [] { "action" },
+					method: "POST",
+					title: "Create",
+					type: "application/x-www-form-urlencoded",
+					fields: new []{ TestHelpers.GetField() }
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					method: "POST",
+					title: "Create",
+					type: "application/x-www-form-urlencoded",
+					fields: new []{ TestHelpers.GetField() }
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					@class: new [] { "action" },
+					title: "Create",
+					type: "application/x-www-form-urlencoded",
+					fields: new []{ TestHelpers.GetField() }
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					@class: new [] { "action" },
+					method: "POST",
+					type: "application/x-www-form-urlencoded",
+					fields: new []{ TestHelpers.GetField() }
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					@class: new [] { "action" },
+					method: "POST",
+					title: "Create",
+					fields: new []{ TestHelpers.GetField() }
+					),
+				new SirenAction(
+					name: "action",
+					href: new Uri( "http://localhost" ),
+					@class: new [] { "action" },
+					method: "POST",
+					title: "Create",
+					type: "application/x-www-form-urlencoded"
+					),
+            };
+        }
+
+		private static TestCaseData[] HashCodeTests()
+        {
+           return HashCodeActions().Select( x => new TestCaseData( x ) ).ToArray() ;
+        }
+
+		private static IEnumerable<TestCaseData> HashCodeEqualityTests() {
+            foreach( var action1 in HashCodeActions() )
+            {
+                var innerEntities = HashCodeActions().ToList();
+				innerEntities.Remove( action1 );
+                foreach (var action2 in innerEntities )
+                {
+                    yield return new TestCaseData( action1, action2 );
+                }
+
+            }
+        }
+
+        [TestCaseSource( nameof( HashCodeTests ) ) ]
+		public void SirenAction_GetHashcodeNot0( ISirenAction action )
+        {

```

Comment:
i know the formatting is off in these files.  i'll fix it after review (also there's no editor config file in this repo :( )